includes/admin/class-wp-statistics-admin-ajax.php

```
@@ -672,6 +695,113 @@ public function dismiss_notices_action_callback()
+
+                wp_send_json_success(['message' => esc_html__('Source channel update successfully.', 'wp-statistics')]);
+            }
+        } catch (Exception $e) {
+            wp_send_json_error(['message' => sprintf(esc_html__('Source channel update failed: %s', 'wp-statistics'), $e->getMessage())]);
+        }
+
+        exit;
+    }
+
+    /**
+     * Handles AJAX requests to anonymize IP addresses of visitors.
+     *
+     * @return void
+     */
+    public function hash_ips_action_callback()
+    {
+        try {
+            if (Request::isFrom('ajax') && User::Access('manage')) {
+                check_ajax_referer('wp_rest', 'wps_nonce');
+
+                $result = IP::Update_HashIP_Visitor();
+
+                wp_send_json_success(['message' => sprintf(esc_html__('Successfully anonymized `%d` IP addresses using hash values.', 'wp-statistics'), $result)]);
+            }
+        } catch (Exception $e) {
+            wp_send_json_error(['message' => sprintf(esc_html__('IP anonymization failed: %s', 'wp-statistics'), $e->getMessage())]);
+        }
+
+        exit;
+    }
+
+    /**
+     * Handles AJAX requests to repair database schema issues for WP Statistics.
+     *
+     * @return void
+     */
+    public function repair_schema_action_callback()
+    {
+        try {
+            if (Request::isFrom('ajax') && User::Access('manage')) {
+                check_ajax_referer('wp_rest', 'wps_nonce');
+
+                $schemaRepairResult = SchemaMaintainer::repair();
+                $databaseStatus     = $schemaRepairResult['status'] ?? null;
+
+                if ($databaseStatus === 'success') {
+                    wp_send_json_success(['message' => esc_html__('Database schema issues have been successfully repaired.', 'wp-statistics')]);
+
+                } else {
+                    wp_send_json_error(['message' => esc_html__('Failed to repair database schema. Please try again.', 'wp-statistics')]);
```

Normally, when there's a `catch` block that handles errors, it's better to just throw an exception and have simpler logic. Something like this:

> if ($databaseStatus !== 'success') {
>     // throw error
> }
> wp_send_json_success(...);
>